tests/tests.h

```diff
@@ -32,6 +32,11 @@
 #include <sstream>
 #include <iomanip>
 
+// silence extra diagnostics in the testsuite
+#ifdef DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
+DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
+#endif
```

Comment:
Nowadays, the testsuite can be configured independently. While it is clear that changes in the library and the testsuite are somewhat tightly coupled, I think it has some value to not break every test when configured against a deal.II library that doesn't provide ```DEAL_II_DISABLE_EXTRA_DIAGNOSTICS```.